java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -1715,12 +1731,26 @@ else if (state.equals("rebooted") && ((VirtualizationRebootGuestAction)action).i
         KickstartableTree tree = KickstartFactory.lookupKickstartTreeByLabel(nameParts.get(0),
                 OrgFactory.lookupById(Long.valueOf(nameParts.get(1))));
         tree.createOrUpdateSaltFS();
-        String kOpts = buildKernelOptions(system, profile, dist, host);
-        Map<String, Object> pillar = new HashMap<>();
+        String kOpts = buildKernelOptions(system, profile, dist, kickstartHost);
+        Map<String, String> pillar = new HashMap<>();
         pillar.put("uyuni-reinstall-kernel", saltFSKernel);
         pillar.put("uyuni-reinstall-initrd", saltFSInitrd);
         pillar.put("uyuni-reinstall-kopts", kOpts);
+
+        return pillar;
```

Comment:
I wonder if we should make these names more general as they are now also used for installing a new instance.
But maybe not in this PR.

Reply:
it's a quick thing to do. Will add it to this PR